core/shared/src/main/scala/scalaz/zio/internal/FiberContext.scala

```diff
@@ -272,6 +273,7 @@ private[zio] final class FiberContext[E, A](
           } else {
             // Interruption cannot be interrupted:
             this.noInterrupt += 1
+            this.terminating = true
```

Comment:
This pattern occurs three times:

```scala
this.noInterrupt += 1
this.terminating = true
```

May be worth defining:

```scala
def beginTermination = {
  this.noInterrupt += 1
  this.terminating = true
}
```

Also, it looks to me like `terminating` is only used transiently, a single time, immediately following the next instruction. As soon as the terminate/interrupt is processed, the stack will be unwound completely, and the flag checked.

If that's true, we can add `FiberStatus.Terminating`, and transition to that state, which will save memory and maybe be clearer.